demo-simple/src/main/java/com/novoda/downloadmanager/demo/VersionOneDatabaseCloner.java

```diff
@@ -77,23 +77,31 @@ private void createFileIfDoesNotExist(File outputFile) {
     }
 
     private void copyAssetToFile(String assetName, File outputFile) {
-        InputStream inputStream;
-        OutputStream myOutput;
+        InputStream inputStream = null;
+        OutputStream myOutput = null;
         int length;
         try {
             inputStream = assetManager.open(assetName);
             myOutput = new FileOutputStream(outputFile, true);
             while ((length = inputStream.read(BUFFER)) > 0) {
                 myOutput.write(BUFFER, 0, length);
             }
-            myOutput.close();
-            myOutput.flush();
-            inputStream.close();
-
             Log.d(getClass().getSimpleName(), "Copied asset: " + assetName);
         } catch (IOException e) {
-            e.printStackTrace();
             Log.e("Failed to copy asset: " + assetName, e);
+        } finally {
+            try {
+                if (myOutput != null) {
```

Comment:
We weren't correctly handling the closing down of the input and output streams before, exposing us to potentially unhandled `IOException`